lib/gst/player/gstplayer-media-info.c

```diff
@@ -0,0 +1,603 @@
+}
+
+static void
+gst_player_video_info_finalize (GObject * gobject)
+{
+  G_OBJECT_CLASS (gst_player_video_info_parent_class)->finalize (gobject);
+}
+
+static void
+gst_player_video_info_class_init (GObjectClass * klass)
+{
+  klass->finalize = gst_player_video_info_finalize;
+}
+
+GstPlayerVideoInfo*
+gst_player_video_info_new (GstPlayerVideoInfo *ref)
+{
+  GstPlayerVideoInfo *ret = NULL;
+
+  ret = g_object_new (GST_TYPE_PLAYER_VIDEO_INFO, NULL);
+
+  if (ref) {
+    ret->width = ref->width;
+    ret->height  = ref->height;
+    ret->framerate_num = ref->framerate_num;
+    ret->framerate_denom = ref->framerate_denom;
+    ret->par_num = ref->par_num;
+    ret->par_denom = ref->par_denom;
+    ret->bitrate = ref->bitrate;
+    ret->max_bitrate = ref->max_bitrate;
+  }
+
+  return ret;
+}
+
+/**
+ * gst_player_video_info_get_width:
+ * @info: a #GstPlayerVideoInfo
+ *
+ * Returns: the width of video in #GstPlayerVideoInfo.
+ */
+guint
+gst_player_video_info_get_width (const GstPlayerVideoInfo *info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), -1);
+
+  return info->width;
+}
+
+/**
+ * gst_player_video_info_get_height:
+ * @info: a #GstPlayerVideoInfo
+ *
+ * Returns: the height of video in #GstPlayerVideoInfo.
+ */
+guint
+gst_player_video_info_get_height (const GstPlayerVideoInfo *info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), -1);
+
+  return info->height;
+}
+
+/**
+ * gst_player_video_info_get_framerate_num:
+ * @info: a #GstPlayerVideoInfo
+ *
+ */
+void
+gst_player_video_info_get_framerate (const GstPlayerVideoInfo *info,
+  guint *fps_n, guint *fps_d)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), NULL);
+
+  *fps_n = info->framerate_num;
+  *fps_d = info->framerate_denom;
+}
+
+/**
+ * gst_player_video_info_get_pixel_aspect_ratio:
+ * @info: a #GstPlayerVideoInfo
+ *
+ */
+void
+gst_player_video_info_get_pixel_aspect_ratio (const GstPlayerVideoInfo *info,
+  guint *par_d, guint *par_n)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), NULL);
+
+  *par_n = info->par_num;
+  *par_d = info->par_denom;
+}
+
+/**
+ * gst_player_video_info_get_bitrate:
+ * @info: a #GstPlayerVideoInfo
+ *
+ * Returns: the current bitrate of video in #GstPlayerVideoInfo.
+ */
+guint
+gst_player_video_info_get_bitrate (const GstPlayerVideoInfo* info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), -1);
+
+  return info->bitrate;
+}
+
+/**
+ * gst_player_video_info_get_max_bitrate:
+ * @info: a #GstPlayerVideoInfo
+ *
+ * Returns: the maximum bitrate of video in #GstPlayerVideoInfo.
+ */
+guint
+gst_player_video_info_get_max_bitrate (const GstPlayerVideoInfo* info)
+{
+  g_return_val_if_fail (GST_IS_PLAYER_VIDEO_INFO (info), -1);
+
+  return info->max_bitrate;
+}
+
+/* Audio information */
+G_DEFINE_TYPE (GstPlayerAudioInfo, gst_player_audio_info,
+    GST_TYPE_PLAYER_STREAM_INFO);
+
+static void
+gst_player_audio_info_init (GstPlayerAudioInfo * info)
+{
+  info->language = NULL;
```

Comment:
It's automatically initialized to NULL, but maybe things like bitrate and samplerate and channels should be initialized to -1